### PR TITLE
WIP: Implement starts_with and ends_with

### PR DESCRIPTION
**Pull request by iamthad**
How does this API look? I tried to model it off the other functions that operate on a rapidstring and take another string argument. It didn't seem to fit in any of the existing sections so I created a new one called "searching".

----
Function declarations, docs, and tests still needed.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -35,7 +35,7 @@ BraceWrapping:
 BreakBeforeBinaryOperators: None
 BreakBeforeBraces: Custom
 BreakBeforeInheritanceComma: false
-BreakBeforeTernaryOperators: false
+BreakBeforeTernaryOperators: true
```
[the remark this answers is not in the document]
Comment (boyerjohn) [reply .clang-format]: Yes, I like this change.

File: include/rapidstring.h [hunk 2]
```diff
@@ -1549,4 +1549,88 @@ RS_API void rs_grow_heap(rapidstring *s, size_t n)
...
+}
+
+RS_API unsigned char rs_ends_with(const rapidstring *s, const char *needle)
+{
+	return rs_ends_with_n(s, needle, strlen(needle));
+}
+
+RS_API unsigned char rs_ends_with_rs(const rapidstring *s, const rapidstring *needle)
+{
+	return RS_DATA_SIZE(rs_ends_with_n, s, needle);
```
[the remark this answers is not in the document]
Comment (boyerjohn) [reply include/rapidstring.h]: hmm, in that case just stick with the macro, if it's not being replaced, might as well use it to not repeat the same function call twice

**Comment by boyerjohn** [comment]
Overall these changes look very nice, I'll gladly merge once the docs and tests are updated.

File: include/rapidstring.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (iamthad) [reply include/rapidstring.h]: The macro doesn't work with a returned value either.

[the remark this answers is not in the document]
Comment (boyerjohn) [reply include/rapidstring.h]: Shoot, nevermind then

**Comment by iamthad** [comment]
I do intend to finish this up when I get a chance and I will open a new PR then.

**Pull request closed without merging**